usvm-core/src/main/kotlin/org/usvm/collection/map/primitive/UMapRegionApi.kt

```diff
@@ -32,6 +32,6 @@ internal fun <MapType, KeySort : USort, ValueSort : USort, Reg : Region<Reg>> UW
         "mapMerge is not applicable to set $region"
     }
 
-    val newRegion = region.merge(srcRef, dstRef, mapType, keySetRegion, guard)
+    val newRegion = region.merge(srcRef, dstRef, mapType, keySetRegion, guard, ownership)
```

Comment:
If we do so, we will have to provide an ownership on `Interpreter` level, and the ownership is assumed to be equal to `memory.ownership`, isn't it?

Reply:
Generally speaking, i don't like that we have an ownership in the memory. Since `memory.write` recieve ownership as an argument we have multiple sources of ownership. Here i see two choices:
1. Remove ownership field from memory. Always use ownership from write argument.
2. Remove ownership argument. Always use ownership from field. Create fresh ownership on `toWritableMemory` (or receive ownership as an argument).

Reply:
For now we don't take ownership as argument in any memory methods, but we have to provide it in `MemoryRegion`, `SymbolicCollection`, `TreeUpdates` (in future).

The existing API is a compromise between uniformity and changing the signatures of a large number of methods